Sources/Apodini/Request/DynamicProperty.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  DynamicProperty.swift
+//  
+//
+//  Created by Max Obermeier on 09.12.20.
+//
+
+import Foundation
+import Runtime
+
+/// `DynamicProperty` allows for wrapping Apodini's property wrappers while maintianing their functionality. By
+/// conforming a `struct` to `DynamicProperty` you this `struct`'s properties discoverable to the Apodini
+/// runtime framework. This can be used to e.g. compine two property wrappers provided by the Apodini framework
+/// into one that combines their functionality
+/// - Precondition: Only structs can be a DynamicProperty
+public protocol DynamicProperty { }
+
+
+// MARK: Execute
+func execute<Element, Target>(_ operation: (Target, _ name: String) -> Void, on element: Element) {
```

Comment:
I would suggest to move all global functions into the `RequestInjectable` or `Injectable` protocol and would suggest to constraint all Injection functionality at one point. I think it is highly beneficial for the whole project if all injection work stays within one point in the project and the shared semantic model builds on top of that functionality. I would suspect that having these methods on a global level could lead to alternative versions of `RequestInjectable` or `Injectable` and therefore lead to more complicated handling in the shared semantic model that should serve as the basis for all the work the different exporters build on top of.

In addition I don't see any usage of `execute` in the current implementation and only see the usage of apply when calling `inject` on a RequestInjectable. Therefore I would suggest to move all the functionality in the inject method of a `RequestInjectable` or `Injectable` protocol. I think this would make it clear for the project that all injectable content must use `RequestInjectable` or `Injectable` and as far as I can see we don't have a specific use case for an alternative for `RequestInjectable` or `Injectable` at the moment, especially after the work that @tgymnich does to decouple `RequestInjectable` or `Injectable` from Vapor.

Reply:
I replaced the remaining usage of `Mirror` in `RequestInjectable.swift` by usage of `execute`. I advise against moving/merging this PR's functionality with `RequestInjectable` or `Injectable` because of one simple reason: I don't think `RequestInjectable` will be able to cover all use-cases on the long run.

I think the shared semantic model is already quite constraining right now. Especially when it comes to downgraded functionality because of constraints from the underlying protocols and the communicational patterns they support in the future, I think some exporters will need more flexibility. This flexibility will most likely include extending Apodini's property wrapper with exporter-specific protocols.

Maybe this can even be unified inside the shared semantic model, using one or two additional protocols, but then we still need `apply` and `execute` as stand-alone functions that are not bound to `RequestInjectable`.